src/Logging/src/DynamicLogger/DynamicLoggingBuilder.cs

```diff
@@ -24,34 +24,69 @@ public static ILoggingBuilder AddDynamicConsole(this ILoggingBuilder builder)
     {
         ArgumentGuard.NotNull(builder);
 
-        // only run if an IDynamicLoggerProvider hasn't already been added
-        if (!builder.Services.Any(sd => sd.ServiceType == typeof(IDynamicLoggerProvider)))
+        if (!IsDynamicLoggerProviderAlreadyRegistered(builder))
         {
-            // remove the original ConsoleLoggerProvider to prevent duplicate logging
-            ServiceDescriptor serviceDescriptor = builder.Services.FirstOrDefault(descriptor => descriptor.ImplementationType == typeof(ConsoleLoggerProvider));
-
-            if (serviceDescriptor != null)
-            {
-                builder.Services.Remove(serviceDescriptor);
-            }
-
-            // make sure logger provider configurations are available
-            if (!builder.Services.Any(descriptor => descriptor.ServiceType == typeof(ILoggerProviderConfiguration<ConsoleLoggerProvider>)))
-            {
-                builder.AddConfiguration();
-            }
+            EnsureLoggerProviderConfigurationsAreAvailable(builder);
+            UpdateConsoleLoggerProviderRegistration(builder.Services);
 
             builder.AddFilter<DynamicConsoleLoggerProvider>(null, LogLevel.Trace);
             builder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<ILoggerProvider, DynamicConsoleLoggerProvider>());
-            builder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<IConfigureOptions<ConsoleLoggerOptions>, ConsoleLoggerOptionsSetup>());
-
-            builder.Services.TryAddEnumerable(ServiceDescriptor
-                .Singleton<IOptionsChangeTokenSource<ConsoleLoggerOptions>,
-                    LoggerProviderOptionsChangeTokenSource<ConsoleLoggerOptions, ConsoleLoggerProvider>>());
-
             builder.Services.AddSingleton(p => p.GetServices<ILoggerProvider>().OfType<IDynamicLoggerProvider>().SingleOrDefault());
+
+            DisableConsoleColorsOnCloudFoundry(builder);
```

Comment:
Are we sure it is actually still needed (for example see: https://github.com/dotnet/runtime/issues/37421) b. Is this needed in K8s/TAP and if so it might be named differently.

Reply:
I think @dtillman mentioned on a call today that it is still needed and is also needed on TAP, so we should probably rename this and switch to `Platform.IsCloudHosted`